react-native-spokestack.d.ts

```diff
@@ -0,0 +1,97 @@
+export = RNSpokestack;
+export as namespace RNSpokestack;
+
+declare namespace RNSpokestack {
+  export function constructor(props?: object): void;
+
+  export enum TraceLevel {
+    DEBUG = 10,
+    PERF = 20,
+    INFO = 30,
+    NONE = 100,
+  }
+
+  export enum TTSFormat {
+    TEXT = 0,
+    SSML = 1,
+    SPEECHMARKDOWN = 2,
+  }
+
+  interface SpokestackRecognizeEvent {
+    transcript: string;
+  }
+
+  interface SpokestackErrorEvent {
+    error: string;
+  }
+
+  interface SpokestackTraceEvent {
+    message: string;
+  }
+
+  interface SpokestackTTSEvent {
+    url: string;
+  }
+
+  interface NLUSlot {
+    type: string;
+    value: string;
+  }
+
+  export interface SpokestackNLUEvent {
+    result: {
+      intent: string;
+      confidence: string;
+      slots: NLUSlot[];
+    };
+  }
+
+  interface SynthesizeOptions {
+    input: string;
+    format: TTSFormat;
+    voice: string;
+  }
+
+  export interface SpokestackConfig {
+    input: string;
+    stages: string[];
```

Comment:
Also curious if these are required or if they have defaults.

Reply:
required, no defaults. at some point (after my current ios work to support them) we can port the `profiles` idea from android (https://spokestack.io/docs/Android/speech-pipeline "How do I set it up?") to allow for simpler config.